File: ios/Scan.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(Scan, NSObject)

- (dispatch_queue_t)methodQueue
{
    return dispatch_get_main_queue();
}

+ (BOOL)requiresMainQueueSetup
{
    return YES;
}

RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(scanImage:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)
@end

File: ios/Scan.swift
import Foundation
import UIKit

@objc(Scan)
class Scan: NSObject {

    @objc(multiply:withB:withResolver:withRejecter:)
    func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        resolve(a*b)
    }
    
    @objc(scanImage:withRejecter:)
    func scanImage(resolve:@escaping RCTPromiseResolveBlock, reject:@escaping RCTPromiseRejectBlock) -> Void {
        let appDelegate = UIApplication.shared.delegate!
        let viewController = (appDelegate.window!?.rootViewController)! as UIViewController
        
        let simpleVC = SimpleViewController(resolve: resolve, reject: reject)
        viewController.present(simpleVC, animated: true, completion: nil)
        
//        let scannerViewController = ImageScannerController(delegate: appDelegate)
//        scannerViewController.modalPresentationStyle = .fullScreen
//        let simpleVC =
    }
}
